### ISSUE TITLE
Nullability-related bug producing BadImageFormatException

### ISSUE THREAD

**Issue by roji** [dotnet/roslyn#38795]
While doing some NRE work on Npgsql, I came across a subtle bug that triggers a BadImageFormatException in runtime. Following is a minimal repro, using dotnet SDK 3.0.100-rc1-014190:

```c#
static class Program
{
    static void Main()
    {
        var x = typeof(Bar<>);
        Console.WriteLine(x);
    }
}

public class Bar<T>
{
    public override bool Equals(object? o) => o is Bar<T> range;
    public override int GetHashCode() => 0;
}
```

Disabling nullability specifically around the Equals method makes the error go away.

The repro is available in https://github.com/roji/CompilerNullabilityBug.

**Comment by roji** [comment]
In case it matters (probably not), this was reproduced on Ubuntu 19.04.

**Comment by jcouv** [comment]
@roji Thanks for reporting this.
This issue was previously reported (https://github.com/dotnet/roslyn/issues/38344) and was fixed in 16.4 preview 1.
I'll go ahead and close as duplicate.

**Comment by roji** [comment]
Thanks @jcouv!
